packages/api-server/src/convert-tx.ts

```diff
@@ -81,50 +80,78 @@ export function calcEthTxHash(encodedSignedTx: HexString): Hash {
   return ethTxHash;
 }
 
-export async function generateRawTransaction(
+/**
+ * Convert the ETH raw transaction to Godwoken L2Transaction
+ */
+export async function ethRawTx2GwTx(
   data: HexString,
   rpc: GodwokenClient
 ): Promise<[L2Transaction, [string, string] | undefined]> {
   logger.debug("convert-tx, origin data:", data);
-  const polyjuiceTx: PolyjuiceTransaction = decodeRawTransactionData(data);
+  const polyjuiceTx: PolyjuiceTransaction = ethRawTx2PolyTx(data);
   logger.debug("convert-tx, decoded polyjuice tx:", polyjuiceTx);
-  const [godwokenTx, cacheKeyAndValue] = await parseRawTransactionData(
+  const [godwokenTx, cacheKeyAndValue] = await polyTx2GwTx(
     polyjuiceTx,
     rpc,
     data
   );
   return [godwokenTx, cacheKeyAndValue];
 }
 
-export function decodeRawTransactionData(
-  dataParams: HexString
-): PolyjuiceTransaction {
-  const result: Buffer[] = rlp.decode(dataParams) as Buffer[];
-  // todo: r might be "0x" which cause inconvenient for down-stream
-  const resultHex = result.map((r) => "0x" + Buffer.from(r).toString("hex"));
-
+/**
+ * RLP decode the ETH raw transaction
+ */
+function rlpDecodeEthTx(ethRawTx: HexString): {
+  nonce: HexString;
+  gasPrice: HexString;
+  gasLimit: HexString;
+  to: HexString;
+  value: HexString;
+  data: HexString;
+  v: HexString;
+  r: HexString;
+  s: HexString;
+} {
+  const result: Buffer[] = rlp.decode(ethRawTx) as Buffer[];
   if (result.length !== 9) {
-    throw new Error("decode raw transaction data error");
+    throw new Error("decode eth raw transaction data error");
   }
 
+  // todo: r might be "0x" which cause inconvenient for down-stream
+  const resultHex = result.map((r) => "0x" + Buffer.from(r).toString("hex"));
   const [nonce, gasPrice, gasLimit, to, value, data, v, r, s] = resultHex;
+  return { nonce, gasPrice, gasLimit, to, value, data, v, r, s };
+}
+
+/**
+ * Convert ETH raw transaction to PolyjuiceTransaction
+ */
+export function ethRawTx2PolyTx(ethRawTx: HexString): PolyjuiceTransaction {
+  const ethTx = rlpDecodeEthTx(ethRawTx);
+  return ethTx2PolyTx(ethTx);
+}
 
+export function ethTx2PolyTx(ethTx: {
+  nonce: HexString;
```

Comment:
Any difference with `PolyjuiceTransaction` ?

Reply:
Done. 1db0157bf7f1d7b2a4b0f9d2f66ddcedf67b5f1a merges functions and removes redundant structure.